### PR TITLE
Add `UnloadEditorTilesInPlayMode` setting

### PR DESCRIPTION
**Pull request by j9liu**
## Description

This PR addresses an issue noted in #1822 re: degraded performance in Play-in-Editor (PIE) mode due to splats being doubly-loaded. Although this PIE behavior is expected, it's undesired when dealing with very intensive datasets.

The `UnloadEditorTilesInPlayMode` setting provides a workaround for this, causing the Editor instance of the tileset to unload itself before PIE starts. Tiles will reload in Editor after PIE ends. Since this could potentially result in consuming more sessions (I think?), I have this setting turned off by default.

## Issue number or link

N/A

## Author checklist

- [X] I have submitted a [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) (only needed once).
- [X] I have done a full self-review of my code.
- [X] I have updated `CHANGES.md` with a short summary of my change (for user-facing changes).
- [ ] ~I have added or updated unit tests to ensure consistent code coverage as necessary.~
  - Unclear how to set up a unit test that switches between Editor and PIE mode. 
- [X] I have updated the documentation as necessary.

## Testing plan

This could be testable with any tileset, but you can measure the impact most prominently with the [Microsoft Campus](https://sandcastle.cesium.com/?id=3d-tiles-gaussian-splats-with-lod) Gaussian splat dataset. Use the Niagara stat panel to compare the number of sprites with and without the setting.

image

## Reviewer checklist

Thank you for taking the time to review this PR. By approving a PR you are taking as much responsibility for these changes as the author.

As you review, please go through the checklist below:

- [ ] Review and run all parts of the test plan on this branch and verify it matches expectations.
    - [ ] If the issue is a bug please make sure you can reproduce the bug in the main branch and then checkout this branch to make sure it actually solved the issue.
- [ ] Review the code and make sure you do not have any remaining questions or concerns. You should understand the code change and the chosen approach. If you are not confident or have doubts about the code, please do not hesitate to ask questions.
    - [ ] Code should follow the [C++ Style Guide](https://cesium.com/learn/cesium-native/ref-doc/style-guide.html)
- [ ] Review the unit tests and make sure there are no missing tests or edge cases.
- [ ] Review documentation changes and updates to `CHANGES.md` to make sure they accurately cover the work in this PR.
- [ ] Verify that the [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) has been submitted, if needed.

### CODE REVIEW

File: Source/CesiumRuntime/Private/Cesium3DTileset.cpp [hunk 1]
```diff
@@ -2436,6 +2447,12 @@ void ACesium3DTileset::RuntimeSettingsChanged(
     this->DestroyTileset();
   }
 }
+
+void ACesium3DTileset::OnPreBeginPIE(bool bIsSimulating) {
```
Comment (david-lively) [inline Source/CesiumRuntime/Private/Cesium3DTileset.cpp:2451]: Doesn't really matter in this implementation, but I believe the parameter should be `const`? Unreal will figure it out regardless.

File: Source/CesiumRuntime/Public/Cesium3DTileset.h [hunk 2]
```diff
@@ -1400,6 +1414,8 @@ class CESIUMRUNTIME_API ACesium3DTileset : public AActor {
   void RuntimeSettingsChanged(
       UObject* pObject,
       struct FPropertyChangedEvent& changed);
+
+  void OnPreBeginPIE(bool bIsSimulating);
```
Comment (david-lively) [inline Source/CesiumRuntime/Public/Cesium3DTileset.h:1418]: See `const` review note on the implementation.

**Review by david-lively — COMMENTED** [review]
I have a few notes, but looks good overall.

File: Source/CesiumRuntime/Private/Cesium3DTileset.cpp (hunk 1 above)
Comment (j9liu) [reply Source/CesiumRuntime/Private/Cesium3DTileset.cpp:2451]: I used to be proactive about adding `const` myself, but our [C++ style guidelines](https://cesium.com/learn/cesium-native/ref-doc/style-guide.html#const) suggest to avoid `const` for by-value parameters. Is that fair? 😅
